apps/prairielearn/src/ee/pages/institutionAdminCourse/institutionAdminCourse.ts

```diff
@@ -39,4 +43,52 @@ router.get(
   }),
 );
 
+const UpdateEnrollmentLimitsBodySchema = z.object({
+  __action: z.literal('update_enrollment_limits'),
+  yearly_enrollment_limit: z.union([z.literal(''), z.coerce.number().int()]),
+  course_instance_enrollment_limit: z.union([z.literal(''), z.coerce.number().int()]),
+});
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    const course = await queryRow(
+      sql.select_course,
+      {
+        institution_id: req.params.institution_id,
+        course_id: req.params.course_id,
+      },
+      CourseSchema,
+    );
+
+    if (req.body.__action === 'update_enrollment_limits') {
+      const body = UpdateEnrollmentLimitsBodySchema.parse(req.body);
+      await runInTransactionAsync(async () => {
+        const updatedCourse = await queryRow(
+          sql.update_enrollment_limits,
+          {
+            course_id: course.id,
+            yearly_enrollment_limit: body.yearly_enrollment_limit || null,
+            course_instance_enrollment_limit: body.course_instance_enrollment_limit || null,
+          },
+          CourseSchema,
+        );
+        await insertAuditLog({
+          authn_user_id: res.locals.authn_user.user_id,
+          table_name: 'pl_courses',
+          action: 'update',
```

Comment:
Not necessarily requesting a change in this PR, but I'd really like to have a column in the audit logs to store some more detail about updates. In PT we use `column_name` for this but I'd be happy with `detail: 'enrollment_limits'` or anything else to help us store what kind of update this was.

Reply:
I'll avoid making any changes in this PR. I think we need a major rethink of audit events in PL anyways.

Reply:
Yep, very fair.